File: Java/Medium/2028.java
class Solution {
    public int[] missingRolls(int[] rolls, int mean, int n) {
        // get the sum of m
        int sumM = 0;
        for (int i = 0; i < rolls.length; i++) {
            sumM += rolls[i];
        }

        int total = mean * (rolls.length + n);

        // check boundaries
        if (total < sumM + n || total > sumM + 6 * n) {
            return new int[0];
        }
        int[] answer = new int[n];

        int sumN = total - sumM;

        // distribute over the average
        for (int j = 0; j < n; j++) {
            answer[j] = sumN / n;
        }

        int remain = sumN % n;
        // distribute the remainder
        for (int k = 0; k < remain; k++) {
            answer[k]++;
        }

        return answer;
    }
}

File: Python/Medium/2028.py
class Solution:
    def missingRolls(self, rolls: List[int], mean: int, n: int) -> List[int]:
        sumM = sum(rolls)
        total = mean * (len(rolls) + n)
        if total < sumM + n or total > sumM + 6 * n:
            return []

        sumN = total - sumM

        answer = [sumN // n] * n
        remain = sumN % n
        for i in range(remain):
            answer[i] += 1

        return answer
